modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -3816,17 +3877,34 @@ def __convert_by(by):
                 return list(by)
             return _convert_by(by)
 
+        drop_column_level = values is not None and not is_list_like(values)
         index, columns, values = map(__convert_by, [index, columns, values])
 
         unique_keys = np.unique(index + columns)
         unique_values = np.unique(values)
 
         if len(values):
-            to_group = self.getitem_column_array(unique_values)
+            to_group = self.getitem_column_array(unique_values, ignore_order=True)
         else:
             to_group = self.drop(columns=unique_keys)
 
-        keys_columns = self.getitem_column_array(unique_keys)
+        keys_columns = self.getitem_column_array(unique_keys, ignore_order=True)
```

Comment:
Can we sort in this function and not introduce `ignore_order` new parameter that only implemented for pandas query compiler?

Reply:
We definitely can, but it will introduce code duplication. I believe that this parameter can be used in more places, as we rarely need the order to be preserved.

> parameter that only implemented for pandas query compiler

I wouldn't say that's a problem. The parameter doesn't force the implementation to do anything specific, it only brings some freedom in how already implemented method can be performed.

Reply:
Okay, maybe we'll say more explicitly then that this option is not yet supported in other query compilers? Something like this:
```python
if ignore_order:
    raise NotImplementedError
```

Reply:
The `ignore_order=True` <i>doesn't force</i> an implementation to ignore the order, it just <i>allows</i> to do so, meaning that if some implementation ignores the `ignore_order=True` parameter then formally the method's contract is not violated as the doc string clearly says:
"Allow returning columns in an arbitrary order for the sake of performance."

Reply:
Sounds logical, however, by reading just the function contract, we won't know if a particular Query Compiler implements this optimization until we look at its implementation. 
Moreover, we can forget to add this optimization, because the code will obviously not highlight this for us.

["Explicit is better than implicit."](https://peps.python.org/pep-0020/#the-zen-of-python)

Reply:
up to you